src/common/bcknd/device/opencl/rhs_maker.c

```diff
@@ -123,18 +123,48 @@ void fluid_makeabf_opencl(void *abx1, void *aby1, void *abz1,
   
 }
 
-void fluid_makebdf_opencl(void *ulag1, void *ulag2, void *vlag1,
+void scalar_rhs_maker_ext_opencl(void *fs_lag, void *fs_laglag, void *fs,
+                                 real *rho, real *ext1, real *ext2,
+                                 real *ext3, int *n) {
+  cl_int err;
+  
+  if (rhs_maker_program == NULL)
+    opencl_kernel_jit(rhs_maker_kernel, (cl_program *) &rhs_maker_program);
+  
+  cl_kernel kernel = clCreateKernel(rhs_maker_program, "scalar_makeext_kernel", &err);
+  CL_CHECK(err);
+
+  CL_CHECK(clSetKernelArg(kernel, 0, sizeof(cl_mem), (void *) &fs_lag));
+  CL_CHECK(clSetKernelArg(kernel, 3, sizeof(cl_mem), (void *) &fs_laglag));
+  CL_CHECK(clSetKernelArg(kernel, 6, sizeof(cl_mem), (void *) &fs));
+  CL_CHECK(clSetKernelArg(kernel, 9, sizeof(real), rho));
+  CL_CHECK(clSetKernelArg(kernel, 10, sizeof(real), ext1));
+  CL_CHECK(clSetKernelArg(kernel, 11, sizeof(real), ext2));
+  CL_CHECK(clSetKernelArg(kernel, 12, sizeof(real), ext3));
+  CL_CHECK(clSetKernelArg(kernel, 13, sizeof(int), n));
```

Comment:
```suggestion
  CL_CHECK(clSetKernelArg(kernel, 0, sizeof(cl_mem), (void *) &fs_lag));
  CL_CHECK(clSetKernelArg(kernel, 1, sizeof(cl_mem), (void *) &fs_laglag));
  CL_CHECK(clSetKernelArg(kernel, 2, sizeof(cl_mem), (void *) &fs));
  CL_CHECK(clSetKernelArg(kernel, 3, sizeof(real), rho));
  CL_CHECK(clSetKernelArg(kernel, 4, sizeof(real), ext1));
  CL_CHECK(clSetKernelArg(kernel, 5, sizeof(real), ext2));
  CL_CHECK(clSetKernelArg(kernel, 6, sizeof(real), ext3));
  CL_CHECK(clSetKernelArg(kernel, 7, sizeof(int), n));
```
must match number of actual arguments